check-approved-requests.py

```diff
@@ -201,13 +201,16 @@ def check_requests(request_type, auth_file, worksheet_key):
                             **request_info)
  
             processed_rows.append(idx)
-            timestamp_spreadsheet(sheet, timestamp, processed_rows)
             if args.log:
                 log_request(args.log, timestamp, request_info['user_email'])
         else:
             # skip over unapproved or already-notified rows
             continue
 
+    # Google API returns an error if you send an empty request
+    if processed_rows:
```

Comment:
Make sense. Just for discussion. Just right off the top of my head, why not use to `try/except` here to catch the non-existed instances. I mean, `if` works perfectly fine here and it might be the best solution for now. But I am just curious.